packages/runtime-dom/src/helpers/useCssVars.ts

```diff
@@ -40,12 +43,32 @@ export function useCssVars(getter: (ctx: any) => Record<string, string>) {
   watchPostEffect(setVars)
 
   onMounted(() => {
-    const ob = new MutationObserver(setVars)
-    ob.observe(instance.subTree.el!.parentNode, { childList: true })
-    onUnmounted(() => ob.disconnect())
+    const obs = [onSubTreeChange(instance.subTree.el!.parentNode, setVars)]
+    const observeTeleportTarget = (vnode: VNode) => {
+      if (vnode.shapeFlag & ShapeFlags.TELEPORT) {
+        const target = resolveTarget(
+          vnode.props as TeleportProps,
+          nodeOps.querySelector
+        ) as Node
+        if (target) {
+          obs.push(onSubTreeChange(target, () => updateCssVars(vnode)))
+        }
+      }
+      if (isArray(vnode.children)) {
+        vnode.children.forEach(n => observeTeleportTarget(n as VNode))
```

Comment:
walk vnode tree and observe all teleport node,there may be a extra performance cost
Indeed,we need further consideration

Reply:
> walk vnode tree and observe all teleport node,there may be a extra performance cost Indeed,we need further consideration

I made some modifications, but as you said, I also think that doing so will consume a lot of performance.
I think a better approach should be like `suspense` , which handles the `cssvars` update on `teleport` at the right time during the `runtime` process, but that is a big change